src/sqlparser/src/ast/mod.rs

```diff
@@ -106,19 +110,57 @@ impl Ident {
         }
     }
 
-    /// Create a new quoted identifier with the given quote and value. This function
-    /// panics if the given quote is not a valid quote character.
-    pub fn with_quote<S>(quote: char, value: S) -> Self
+    /// Create a new identifier from word
+    pub fn new_from_word(w: &Word) -> Result<Ident, ParserError> {
+        if w.value.is_empty() {
+            Err(ParserError::ParserError(format!(
+                "zero-length delimited identifier at or near \"{w}\""
+            )))
+        } else {
+            Ok(Ident {
+                value: w.value.clone(),
+                quote_style: w.quote_style,
+            })
+        }
+    }
+
+    /// Create a new quoted identifier with the given quote and value.
+    /// the inputs are guanteed to be legal.
+    pub fn new_with_quote_safe<S>(quote: char, value: S) -> Self
     where
         S: Into<String>,
     {
-        assert!(quote == '\'' || quote == '"' || quote == '`' || quote == '[');
         Ident {
             value: value.into(),
             quote_style: Some(quote),
         }
     }
 
+    /// Create a new quoted identifier with the given quote and value.
+    /// returns ParserError when the given string is empty or the given quote is illegal.
+    pub fn new_with_quote_check<S>(quote: char, value: S) -> Result<Ident, ParserError>
```

Comment:
Where is this function used? I didn't find that

Reply:
shoud I remove this function? it might be useful, and I just found this asymmetry jarring defining only a unchecked function